packages/integration/src/gitlab/core.ts

```diff
@@ -101,20 +105,27 @@ export function buildRawUrl(target: string): URL {
 // Converts
 // from: https://gitlab.com/groupA/teams/teamA/subgroupA/repoA/-/blob/branch/filepath
 // to:   https://gitlab.com/api/v4/projects/projectId/repository/files/filepath?ref=branch
-export function buildProjectUrl(target: string, projectID: Number): URL {
+export function buildProjectUrl(
+  target: string,
+  projectID: Number,
+  config: GitLabIntegrationConfig,
+): URL {
   try {
     const url = new URL(target);
 
     const branchAndFilePath = url.pathname.split('/-/blob/')[1];
     const [branch, ...filePath] = branchAndFilePath.split('/');
-
-    url.pathname = [
-      '/api/v4/projects',
-      projectID,
-      'repository/files',
-      encodeURIComponent(decodeURIComponent(filePath.join('/'))),
-      'raw',
-    ].join('/');
+    const relativePath = getGitLabIntegrationRelativePath(config);
+
+    url.pathname =
+      [relativePath] +
+      [
```

Comment:
Whoa :) adding two arrays isn't a thing in javascript. This code actually accidentally works out in practice because of how it stringifies both sides of the expression, but this needs to be fixed. Maybe something like

```
[
  ...(relativePath ? [relativePath] : []),
  '/api/v4/projects',
  <etc>
].join('/')
```